internal/analytics/analytics.go

```diff
@@ -15,6 +16,16 @@ type Dispatcher interface {
 	Close()
 }
 
+var AnalyticsURL string
+
 func init() {
 	configMediator.RegisterOption(constants.ReportAnalyticsConfig, configMediator.Bool, true)
+	configMediator.RegisterOption(constants.AnalyticsPixelOverrideConfig, configMediator.String, "")
+}
+
+func RegisterConfigListener(cfg *config.Instance) error {
+	configMediator.AddListener(constants.AnalyticsPixelOverrideConfig, func() {
```

Comment:
Far as I can tell this does not speak with the state-svc. So a listener placed on the svc will not fire if the config was changed on the state tool side.

I like this pattern though, and it seems to me we ought to support this across state tool and the svc. It also means the code as is is completely self sufficient aside from the state-svc AC, but if we address that separately this code will automatically get the benefit.

I've filed this as a followup: https://activestatef.atlassian.net/browse/CP-1083, and we'll drop the AC from the current set of stories regarding the svc picking up the change. I don't think it's critical for the milestone we're targeting.

Good thinking on taking this approach!

Reply:
I'm _fairly_ certain that the service will pick up this change. When we set a config value the runner [makes a request to the service](https://github.com/ActiveState/cli/blob/master/internal/runners/config/set.go#L90) which in turn [notifies the listeners in the service](https://github.com/ActiveState/cli/blob/master/cmd/state-svc/internal/resolver/resolver.go#L259).

Reply:
Ohh nice! I wasn't grokking that yesterday. Thanks for digging! 🙏